src/telemetry/telemetry.ts

```diff
@@ -43,7 +45,17 @@ class Telemetry implements ITelemetry {
 
   private filterFunction(envelope: any) {
     // Identifies the source of telemetry events
-    envelope.baseData.name = 'Graph Explorer v4';
+    envelope.tags['ai.cloud.role'] = 'Graph Explorer v4';
+
+    // Redact PII from query url
+    const properties = envelope.baseData.properties || {};
+    for (const property in properties) {
+      if (property === 'QueryUrl')
```

Comment:
We shouldn't be doing this. I thought there was agreement that we would stay away from URLs and req/resp bodies. 

First, it is impossible for us to know all potential PII in URLs exposed by the different service APIs. Even if we nail it today, we don't review all new URLs so it is very possible that new forms of PII come onboard. So, take ID format. That format appears to be a guid. There is no guarantee that all identifiers or GUIDs. For example, OneDrive creates folder identifiers like:
`01DVF26FAOKNM7R54BENH3CL6W4KKY5MCL`. I have no idea how you regex for that.

Or OneDrive can create file names like:
`PUT /me/drive/root:/FolderA/FileB.txt:/content` <-- This case we can regex the content between the `:`, but I could imagine other scenarios where we don't have tokens to use for regex matches.

Second, we will need to support other requirements for the eventuality that we capture PII. There are significant requirements here IIRC for purging this information, protecting it, etc.

I feel strongly that we shouldn't capture URLs and depend on telemetry sanitization. With that, I think it is safe to capture URLs up through the entity set or singleton only if we validate that the user has entered a valid top level entityset or singleton. We'd want to validate as a user could conceivably enter PII by accident as the entity set. So, we could capture `https://graph.microsoft.com/{endpoint}/{entityset}/{MASKED*}` excluding the masked part.

Reply:
I'll remove URL from telemetry data. Let's talk some more on this.